### ISSUE TITLE
History search can start with what's written in the prompt

### ISSUE THREAD

**Issue by strboul** [randy3k/radian#147]
In *radian*, we have a nice history search with  <kbd>CTRL</kbd>+<kbd>r</kbd>; however, the search doesn't take what has been put after the prompt. 

Hard to describe so it's better to give an example: A user starts writing a piece of code, which is executed in the past and it's in the history, and does a search on the midway but the search prompt is empty and the user has to type everything from the beginning for the completion. 

It'd be nice to have an option for it.

**Comment by randy3k** [comment]
Try using the up arrow in this case then.

**Comment by strboul** [comment]
![](https://user-images.githubusercontent.com/25015317/74568630-b33cd680-4f78-11ea-9928-ee06f9faee34.gif)

The case is still something else. For example:

1. I write "inst" in the console
2. Then press CTRL + r
3. Then the search prompt initializes empty but it should have contained what I wrote after the prompt

This nice feature is already implemented in zsh and I believe it'll be a nice addition to *radian* as well.

**Comment by randy3k** [comment]
I agree it’s a nice feature. I just want to emphasis there is a workaround now

1. Type inst 
2. Hide the completion menu by ctrl + c
3. Press the up arrow

**Comment by randy3k** [comment]
Btw, how did you enable such feature in zsh?

**Comment by strboul** [comment]
I am not sure about this option because use zsh with [fzf](https://github.com/junegunn/fzf) but I've just found a similar discussion [here](https://unix.stackexchange.com/questions/97843/how-can-i-search-history-with-text-already-entered-at-the-prompt-in-zsh).

**Comment by randy3k** [comment]
The substring search is what we have implemented with the up arrow key.
